Add product docs for Studio and migration (#24)
extend-hq/documentation: product-reference/studio/configuring_extraction.mdx

> ---

Reviewer: Not related to your PR, but what is the intended difference between the `Documentation` and `Product Reference` tabs? I feel like we should merge the two so that the main docs tab isn't so empty.

Author: I think it's like super legacy from when this initial repo was created and was just the default tab created by Mintlify template, agree we should remove, and api-reference should probably be primary page

Author: I can do that

Author: actually it feels pretty weird when doing it that way

> title: "Configure an extraction processor"

Reviewer: ```suggestion
title: "Configure an Extraction processor"
```

description: "How to configure an Extraction processor in the Extend Studio"
---

## Builder

> Once you have created an Extraction processor, you can navigate to the processor detail view and select the "Builder" tab.

Reviewer: ```suggestion
Once you have created an Extraction processor, you can navigate to the processor detail view and select the "Build" tab.
```

![](/images/studio/extractor_builder.png)

If you have already configured extraction steps in the past in our legacy workflow extraction editor, this will look and feel similar.

> You can save changes without running or save and run using the "Save and run extractor" button. The results will then show up below.

Reviewer: ```suggestion
You can save changes without running or save and run using the "Save and run extraction" button. The results will then show up below.
```

![](/images/studio/extractor_results.png)

## Fields

> To configure a field, add a semantically acurate field name and write descriptions that explain how to identify and extract that field from the document.

Reviewer: ```suggestion
To configure a field, add a semantically accurate field name and write a description that explain how to identify and extract that field from the document.
```

You must also configure the proper field type:

![](/images/studio/field_configure.png)

### Text

Use the text data type when you want to extract a string of text from a document. For example, if you want to extract the name of a person from a document, you would use the text data type.

### Number

Use the number data type when you want to extract a number from a document. For example, if you want to extract the age of a person from a document, you would use the number data type.

### Currency

> Use the currency data type when you want to extract a currency value from a document. For example, if you want to extract the price of a product from a document, you would use the currency data type.

Reviewer: Should we mention anything about the currency, i.e. USD vs EUR?

Signature fields will automatically extract all relevant details of a document's signature block:
- is_signed
- printed_name
- signatory_title
- signature_date

### Object

Use the object data type when you want to extract a set of related fields from a document. For example, if you want to extract the address, name, and birth date of a person from a document you would use the object data type.

### Array

Use the array data type when you want to extract a list of related fields from a document. For example, if you want to extract a list of products that each have a name, price, and quantity from a document you would use the array data type.

## Configuration table

The field config table also will allow you to select the drag button to move the field up or down. Performance is best when related fields in the document
are positioned in related order in the configuration table.

You can also set a field `ID` which is a unique identifier for the field to use in your downstream system, so that you can make changes to the semantic field name without 
updating your downstream system.

![](/images/studio/field_control.png)

## Configuring Custom Settings

> In addition to the fields, you can also configure custom settings for each field. These settings allow you to furhter customize the extraction process to better suit your specific needs. However, please note that these settings are experimental and may not work as expected in all cases.

Reviewer: ```suggestion
In addition to the fields, you can also configure custom settings for each field. These settings allow you to further customize the extraction process to better suit your specific needs. However, please note that these settings are experimental and may not work as expected in all cases.
```

Before using these settings, we recommend consulting with the Extend team to understand their potential impact on the extraction process.

> # Using the runner tab

Reviewer: ```suggestion
# Using the Run tab
```

> While it often makes sense to run files from the Builder ui when getting set up, once you are ready to start testing your processor at scale, you should move over to the Run tab.

Reviewer: ```suggestion
While it often makes sense to run files from the "Build" tab when getting set up, once you are ready to start testing your processor at scale, you should move over to the "Run" tab.
```

From this tab you can:
- Quickly run any number of files in a batch

> - Select the version of the processor you want to run (or default to draft of current saved changes)

Reviewer: ```suggestion
- Select the version of the processor you want to run (or default to the saved draft version)
```

- Run an existing [Evaluation set](/product-reference/evaluation/using_evaluation_sets) for the processor

![](/images/studio/runner.png)

Once you run a batch of files, you will be redirected to a results page that looks like this:

![](/images/studio/batch_result.png)

From here you can:
- See at a glance the coverage of fields extracted and average confidence
- Drill down into individual files to see the extracted fields and confidence levels
- (optionally) correct/edit the results of each output, then turn the entire batch into an Evaluation set

> Note: our recommendation is to not create Evaluation Sets until you have at least mostly finalized what fields you are extracting, even if you are still iterating on the field descriptions.

Reviewer: It might be nice to add an extra sentence to explain the rationale for this recommendation.

# Publishing

> Once the change you are making has been iterated on enough, you are ready to Publish a new version.

Reviewer: It's a bit ambiguous what "iterated on enough" means. Maybe we can replace this with a statement about changes being saved to draft and when you would want to publish a new version.

Reviewer: Actually I see the publishing page goes into this more, maybe we don't need this section?
